pinot-common/src/test/java/org/apache/pinot/common/request/RequestContextUtilsTest.java

```diff
@@ -0,0 +1,191 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.request;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.pinot.common.request.context.ExpressionContext;
+import org.apache.pinot.common.request.context.FilterContext;
+import org.apache.pinot.common.request.context.FunctionContext;
+import org.apache.pinot.common.request.context.RequestContextUtils;
+import org.apache.pinot.common.request.context.predicate.EqPredicate;
+import org.apache.pinot.segment.spi.IndexSegment;
+import org.apache.pinot.segment.spi.SegmentMetadata;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.apache.pinot.segment.spi.index.mutable.ThreadSafeMutableRoaringBitmap;
+import org.apache.pinot.segment.spi.index.startree.StarTreeV2;
+import org.apache.pinot.spi.data.readers.GenericRow;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class RequestContextUtilsTest {
+  private IndexSegment _indexSegment = new IndexSegment() {
+    @Override
+    public String getSegmentName() {
+      return null;
+    }
+
+    @Override
+    public SegmentMetadata getSegmentMetadata() {
+      return null;
+    }
+
+    @Override
+    public Set<String> getColumnNames() {
+      return null;
+    }
+
+    @Override
+    public Set<String> getPhysicalColumnNames() {
+      return ImmutableSet.of("$ts$MONTH");
+    }
+
+    @Override
+    public DataSource getDataSource(String columnName) {
+      return null;
+    }
+
+    @Override
+    public List<StarTreeV2> getStarTrees() {
+      return null;
+    }
+
+    @Nullable
+    @Override
+    public ThreadSafeMutableRoaringBitmap getValidDocIds() {
+      return null;
+    }
+
+    @Override
+    public GenericRow getRecord(int docId, GenericRow reuse) {
+      return null;
+    }
+
+    @Override
+    public void destroy() {
+
+    }
+  };
+
+  @Test
+  public void testExpressionContextHashcode() {
+    ExpressionContext expressionContext1 = ExpressionContext.forIdentifier("abc");
+    ExpressionContext expressionContext2 = ExpressionContext.forIdentifier("abc");
+    Assert.assertEquals(expressionContext1, expressionContext2);
+    Assert.assertEquals(expressionContext1.hashCode(), expressionContext2.hashCode());
```

Comment:
were these not equal before?

Reply:
The enum hashcode from the different JVMs is different. So the hash solution works in quickstart when deploying broker and server in same JVM, but doesn't work in a real prod environment.

Reply:
But the hashCode should *never* leave the JVM, so why do we care?

Reply:
The previous version sends over the hashcode of ExpressionContext from broker to servers and expects exactly the same hashcode in the Pinot server for expression override, hence the problem, which is invalid now.